src/main/rx/ghst.c

```diff
@@ -0,0 +1,312 @@
+    }
+
+    if (shouldSendTelemetryFrame()) {
+        return RX_FRAME_PROCESSING_REQUIRED;
+    }
+
+    return RX_FRAME_PENDING;
+}
+
+static bool ghstProcessFrame(const rxRuntimeConfig_t *rxRuntimeConfig)
+{
+    // Assume that the only way we get here is if ghstFrameStatus returned RX_FRAME_PROCESSING_REQUIRED, which indicates that the CRC
+    // is correct, and the message was actually for us.
+
+    // do we have a telemetry buffer to send?
+    if (shouldSendTelemetryFrame()) {
+        ghstTransmittingTelemetry = true;
+        ghstRxSendTelemetryData();
+    }
+
+    if (ghstValidatedFrameAvailable) {
+        int startIdx = 0;
+
+        if (
+            ghstValidatedFrame.frame.type >= GHST_UL_RC_CHANS_HS4_FIRST && 
+            ghstValidatedFrame.frame.type <= GHST_UL_RC_CHANS_HS4_LAST
+        ) {
+            const ghstPayloadPulses_t* const rcChannels = (ghstPayloadPulses_t*)&ghstValidatedFrame.frame.payload;
+
+            // all uplink frames contain CH1..4 data (12 bit)
+            ghstChannelData[0] = rcChannels->ch1to4.ch1 >> 1;
+            ghstChannelData[1] = rcChannels->ch1to4.ch2 >> 1;
+            ghstChannelData[2] = rcChannels->ch1to4.ch3 >> 1;
+            ghstChannelData[3] = rcChannels->ch1to4.ch4 >> 1;
+
+            switch(ghstValidatedFrame.frame.type) {
+                case GHST_UL_RC_CHANS_HS4_RSSI: {
+                    const ghstPayloadPulsesRSSI_t* const rssiFrame = (ghstPayloadPulsesRSSI_t*)&ghstValidatedFrame.frame.payload;
+                    lqTrackerSet(rxRuntimeConfig->lqTracker, scaleRange(constrain(rssiFrame->lq, 0, 100), 0, 100, 0, RSSI_MAX_VALUE));
+                    
+                    break;
+                }
+
+                case GHST_UL_RC_CHANS_HS4_5TO8:     startIdx = 4;  break;
+                case GHST_UL_RC_CHANS_HS4_9TO12:    startIdx = 8;  break;
+                case GHST_UL_RC_CHANS_HS4_13TO16:   startIdx = 12; break;
+            }
+
+            if (startIdx > 0)
+            {
+                // remainder of uplink frame contains 4 more channels (8 bit), sent in a round-robin fashion
+
+                ghstChannelData[startIdx++] = rcChannels->cha << 3;
+                ghstChannelData[startIdx++] = rcChannels->chb << 3;
+                ghstChannelData[startIdx++] = rcChannels->chc << 3;
+                ghstChannelData[startIdx++] = rcChannels->chd << 3;
+            }
+        }
+    }
+
+    return true;
+}
+
+STATIC_UNIT_TESTED uint16_t ghstReadRawRC(const rxRuntimeConfig_t *rxRuntimeState, uint8_t chan)
+{
+    UNUSED(rxRuntimeState);
+
+    // derived from original SBus scaling, with slight correction for offset (now symmetrical
+    // around OpenTx 0 value)
+    // scaling is:
+    //      OpenTx   RC     PWM
+    // min  -1024    172    988us
+    // ctr  0        992    1500us
+    // max  1024     1811   2012us
+    //
+
+    return (5 * (ghstChannelData[chan]+1) / 8) + 880;
```

Comment:
This is super similar to S.Bus. Can we reuse `sbusDecodeChannelValue()` here? Stick jitter is likely to be larger than 5/8 usec anyway.